xdsl/dialects/memref.py

```diff
@@ -492,6 +493,8 @@ class AlterShapeOp(IRDLOperation):
         "$src $reassociation attr-dict `:` type($src) `into` type($result)"
     )
 
+    traits = frozenset([NoMemoryEffect()])
+
```

Comment:
Not sure about this one as I see no reference to it upstream, but it sounds sensible given the name :shrug:

Reply:
These are marked `NoMemoryEffect` also in upstream MLIR: see both [memref.collapse_shape](https://mlir.llvm.org/docs/Dialects/MemRef/#memrefcollapse_shape-memrefcollapseshapeop) and [memref.expand_shape](https://mlir.llvm.org/docs/Dialects/MemRef/#memrefexpand_shape-memrefexpandshapeop)